dt: Fixups for CDT FIPS

michael-redpanda opened the pull request:
FIPS CDT is failing because the admin password is too short and the rdkafka based clients (which use OpenSSL in FIPS mode) are failing to construct their final client SCRAM message.  This commit updates the superuser password to be at least 14 characters long.

This PR does the following:

* Updates the superuser password to be at least 14 characters long
* Updates some passwords used in tests that were too short causing those tests to fail
* Add `skip_fips` to relevant tests

TODO:
- [x] Run CDT FIPS

Fixes: CORE-14533
Fixes: CORE-8217

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [X] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.3.x
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

* None

michael-redpanda committed:
dt: Update admin password

FIPS CDT is failing because the admin password is too short and the
rdkafka based clients (which use OpenSSL in FIPS mode) are failing to
construct their final client SCRAM message.  This commit updates the
superuser password to be at least 14 characters long.

Signed-off-by: Michael Boquard <michael@redpanda.com>

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--779bfc22-5306-4038-90f3-9d5638324e08-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/76746">build#76746</a></summary>


| test_class                          | test_method                | test_arguments                      | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                              |
|:------------------------------------|:---------------------------|:------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| AuditLogTestKafkaApi                | test_no_auth_enabled       | {"audit_transport_mode": "kclient"} | integration | https://buildkite.com/redpanda/redpanda/builds/76746#019aa285-1829-4587-ad34-e0f71c43edcb | FLAKY         | 19/21    | upstream reliability is '93.0062630480167'. current run reliability is '90.47619047619048'. drift is 2.53007 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=AuditLogTestKafkaApi&test_method=test_no_auth_enabled                   |
| MountUnmountIcebergTest             | test_simple_remount        | {"cloud_storage_type": 1}           | integration | https://buildkite.com/redpanda/redpanda/builds/76746#019aa285-1823-40f6-aca5-15ae914c3170 | FLAKY         | 17/21    | upstream reliability is '78.5472972972973'. current run reliability is '80.95238095238095'. drift is -2.40508 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=MountUnmountIcebergTest&test_method=test_simple_remount                 |
| ScalingUpTest                       | test_fast_node_addition    | null                                | integration | https://buildkite.com/redpanda/redpanda/builds/76746#019aa285-1822-4543-863e-8cf71a492d0e | FLAKY         | 18/21    | upstream reliability is '98.34983498349835'. current run reliability is '85.71428571428571'. drift is 12.63555 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ScalingUpTest&test_method=test_fast_node_addition                       |
| TombstoneRemovalTest                | test_single_slow_follower  | null                                | integration | https://buildkite.com/redpanda/redpanda/builds/76746#019aa28c-42f5-4f45-a035-0f4842500a66 | FLAKY         | 8/9      | upstream reliability is '99.36507936507937'. current run reliability is '88.88888888888889'. drift is 10.47619 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TombstoneRemovalTest&test_method=test_single_slow_follower              |
| WriteCachingFailureInjectionE2ETest | test_crash_all             | {"use_transactions": false}         | integration | https://buildkite.com/redpanda/redpanda/builds/76746#019aa285-1826-4e84-a546-49ad19fd7a03 | FLAKY         | 20/21    | upstream reliability is '91.06029106029106'. current run reliability is '95.23809523809523'. drift is -4.1778 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=WriteCachingFailureInjectionE2ETest&test_method=test_crash_all          |
| WriteCachingFailureInjectionTest    | test_unavoidable_data_loss | null                                | integration | https://buildkite.com/redpanda/redpanda/builds/76746#019aa285-1822-4543-863e-8cf71a492d0e | FLAKY         | 18/21    | upstream reliability is '95.28178243774575'. current run reliability is '85.71428571428571'. drift is 9.5675 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=WriteCachingFailureInjectionTest&test_method=test_unavoidable_data_loss |
</details>
<!--/body-->

michael-redpanda commented:
/cdt
fips

michael-redpanda committed:
dt: Add skip_fips_mode to more audit log tests

Since we are unable to build the OCSF server on a FIPS node, skip all
audit log tests.

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
dt: Update more passwords

Updates passwords for tests that are too short to work on FIPS nodes

Signed-off-by: Michael Boquard <michael@redpanda.com>

IoannisRP reviewed (approved):
I like the new passwords 😆

BenPope reviewed (approved):
> Since we are unable to build the OCSF server on a FIPS node

Interesting. For posterity, why is this?

michael-redpanda commented:
> > Since we are unable to build the OCSF server on a FIPS node
> 
> Interesting. For posterity, why is this?

Because it won't build on a FIPS broker.  No idea :( Some issue with crypto

michael-redpanda commented:
> > > Since we are unable to build the OCSF server on a FIPS node
> > 
> > 
> > Interesting. For posterity, why is this?
> 
> Because it won't build on a FIPS broker. No idea :( Some issue with crypto

IIRC - some issue with how the elixir build fetches packages and verifies them.

michael-redpanda commented:
Actually I wonder if with the change in AMI it may work again - i'll have to investigate

michael-redpanda commented:
/cdt
fips

michael-redpanda commented:
There are still failures in CDT but this significantly cleans up (>300 down to a dozen).
